### PR TITLE
Fixes #22096 - Make Audits taxable

### PR DESCRIPTION
**Pull request by tbrisker**
This relies on https://github.com/collectiveidea/audited/pull/413 for taxonomies to be recorded properly on destroy.
Still missing tests as well.

### CODE REVIEW

**Review by dLobatog — DISMISSED** [review]
Setting to WoC because of tests, the feature works fine, thanks @tbrisker !

File: app/models/taxonomy.rb [hunk 1]
```diff
@@ -154,8 +154,8 @@ def dup
     new
   end
 
-  # overwrite *_ids since need to check if ignored? - don't overwrite location_ids and organizations_ids since these aren't ignored
-  (TaxHost::HASH_KEYS - [:location_ids, :organizations_ids]).each do |key|
+  # overwrite *_ids since need to check if ignored? - don't overwrite location_ids and organization_ids since these aren't ignored
+  (TaxHost::HASH_KEYS - [:location_ids, :organization_ids]).each do |key|
```
Comment (tbrisker) [inline app/models/taxonomy.rb:158]: This is an unrelated bug discovered due to the check for `respond_to?(organization_ids)`

File: test/models/concerns/audit_extensions_test.rb [hunk 2]
```diff
@@ -57,4 +57,45 @@ def setup
...
+      domain.locations = [@loc]
+      domain.organizations = [@org]
+      domain.save!
+      audit = domain.audits.last
+      assert_equal 'update', audit.action
+      assert_equal [@loc.id], audit.location_ids
+      assert_equal [@org.id], audit.organization_ids
+    end
+
+    test 'records on destroy' do
```
Comment (tbrisker) [inline test/models/concerns/audit_extensions_test.rb:92]: This test should fail until a new audited version is released containing collectiveidea/audited#413

**Comment by tbrisker** [comment]
leaving on WoC until new audited version is released

File: test/models/setting_test.rb [hunk 3]
```diff
@@ -204,11 +204,12 @@ def test_value_is_updated_after_change_in_SETTINGS
     setting_name = "foo_#{rand(1000000)}"
     Setting.create!(:name => setting_name, :value => "bar", :default => "default", :description => "foo")
 
-    SETTINGS.stubs(:key?).with(setting_name.to_sym).returns(true)
-    SETTINGS.stubs(:[]).with(setting_name.to_sym).returns("no-bar")
+    SETTINGS[setting_name.to_sym] = "no-bar"
```
Comment (tbrisker) [inline test/models/setting_test.rb:207]: These stubs failed because creating the setting checked if taxonomies are enabled for the audit, which the stub didn't expect. However, `SETTINGS` is just a hash, so stubbing it is actually slower then just updating it.

**Comment by tbrisker** [comment]
new audited version cause seeds test to fail, investigating.

**Comment by mmoll** [comment]
[test upgrade]

**Comment by mmoll** [comment]
[test foreman]

**Comment by ares** [comment]
I did a quick test with https://github.com/theforeman/foreman/pull/5332 and it works reasonably well. It does not show the audit in the organization from which I'm removing organization from which can be considered a bug, @rplevka already mentioned that as well. But if that's hard to solve, I'd be fine keeping that for separate issue, as this is already a great improvement that we should deliver in 1.18.

EDIT:
forgot to mention, this would be needed too, since Taxonomix will already need audit DSL in model
```diff
diff --git a/config/initializers/audit.rb b/config/initializers/audit.rb
index 8f276e6f0..034ce7c63 100644
--- a/config/initializers/audit.rb
+++ b/config/initializers/audit.rb
@@ -2,11 +2,12 @@
 # we reopen here to add search functionality
 require 'audited'
 
-Audit = Audited::Audit
-Audit.send(:include, AuditExtensions)
-
 # Re-opened AuditorInstanceMethods to audit 1-0-* associations
 Auditor = Audited::Auditor::AuditedInstanceMethods
 Auditor.send(:include, AuditAssociations)
 
 ::ActiveRecord::Base.send :include, AuditAssociations::Auditor
+
+# Audit includes Taxonomix which already relies on DSL provided by audited gem
+Audit = Audited::Audit
+Audit.send(:include, AuditExtensions)
```

**Review by ares — APPROVED** [review]
this one gets in first, so no further changes are needed, thanks @tbrisker, mering
